ext/clang_extraction/JLLEnvs.jl

```diff
@@ -0,0 +1,240 @@
+module JLLEnvs
+
+using Pkg
+using Pkg.Artifacts
+using Downloads
+
+get_arch_os_libc(target::AbstractString) = get_arch(target), get_os(target), get_libc(target)
+
+get_arch(target::AbstractString) = first(split(target, '-'))
+
+function get_os(target::AbstractString)
+    _, vendor, _ = split(target, '-')
+    if vendor == "apple"
+        os = "macos"
+    elseif vendor == "w64"
+        os = "windows"
+    elseif vendor == "unknown"
+        os = "freebsd"
+    elseif vendor == "linux"
+        os = "linux"
+    else
+        error("Unknown OS: $target")
+    end
+    return os
+end
+
+function get_libc(target::AbstractString)
+    _, _, env = split(target, '-')
+    if startswith(env, "gnu")
+        return "glibc"
+    elseif startswith(env, "musl")
+        return "musl"
+    else
+        return ""
+    end
+end
+
+const JLL_ENV_SHARDS_URL = "https://raw.githubusercontent.com/JuliaPackaging/BinaryBuilderBase.jl/master/Artifacts.toml"
+const JLL_ENV_SHARDS = Dict{String,Any}()
+
+function __init__()
+    merge!(JLL_ENV_SHARDS, Artifacts.load_artifacts_toml(Downloads.download(JLL_ENV_SHARDS_URL)))
+end
+
+const JLL_ENV_HOST_TRIPLE = "x86_64-linux-musl"
+const JLL_ENV_GCC_SHARD_NAME = "GCCBootstrap"
+const JLL_ENV_SYSTEM_SHARD_NAME = "PlatformSupport"
+
+const JLL_ENV_GCC_VERSIONS = VersionNumber[
+    v"4.8.5",
+    v"5.2.0",
+    v"6.1.0",
+    v"7.1.0",
+    v"8.1.0",
+    v"9.1.0",
+    v"10.2.0",
+    v"11.0.0-iains",
+]
+
+const JLL_ENV_TRIPLES = String[
+    "aarch64-apple-darwin20",
+    "aarch64-linux-gnu",           # Tier 1
+    "aarch64-linux-musl",
+    "armv7l-linux-gnueabihf",
+    "armv7l-linux-musleabihf",
+    "i686-linux-gnu",              # Tier 1
+    "i686-linux-musl",
+    "i686-w64-mingw32",            # Tier 2
+    "powerpc64le-linux-gnu",
+    "x86_64-apple-darwin14",       # Tier 1
+    "x86_64-linux-gnu",            # Tier 1
+    "x86_64-linux-musl",
+    "x86_64-unknown-freebsd11.1",  # Tier 1
+    "x86_64-w64-mingw32",          # Tier 1
+]
```

Comment:
You can load `BinaryBuilderBase` at this point?

Reply:
This is copied from https://github.com/JuliaInterop/Clang.jl/blob/master/src/shards/JLLEnvs.jl. I wrote this because I don't wanna add `BinaryBuilderBase` as a dependency of Clang.jl, but I think it's ok to use `BinaryBuilderBase` for this project.